dashboard/app/controllers/pd/mail_preview_controller.rb

```diff
@@ -0,0 +1,159 @@
+class Pd::MailPreviewController < MailView
+  # Only in development environments, allows previewing mail rendering with fixed test data
+
+  # Teacher enrollment receipt
+  # CSF
+  def teacher_enrollment_receipt_csf
+    workshop = workshop(Pd::Workshop::COURSE_CSF)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  # CSinA
+  def teacher_enrollment_receipt_cs_in_a_phase_2
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_A, subject: Pd::Workshop::SUBJECT_CS_IN_A_PHASE_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_cs_in_a_phase_3
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_A, subject: Pd::Workshop::SUBJECT_CS_IN_A_PHASE_3)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  # CSinS
+  def teacher_enrollment_receipt_cs_in_s_phase_2
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_cs_in_s_phase_3_semester_1
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_3_SEMESTER_1)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_cs_in_s_phase_3_semester_2
+    workshop = workshop(Pd::Workshop::COURSE_CS_IN_S, subject: Pd::Workshop::SUBJECT_CS_IN_S_PHASE_3_SEMESTER_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  # ECS
+  def teacher_enrollment_receipt_ecs_phase_2
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_PHASE_2)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_3
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_3)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_4
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_4)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_5
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_5)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_unit_6
+    workshop = workshop(Pd::Workshop::COURSE_ECS, subject: Pd::Workshop::SUBJECT_ECS_UNIT_6)
+    Pd::WorkshopMailer.teacher_enrollment_receipt enrollment(workshop)
+  end
+
+  def teacher_enrollment_receipt_ecs_phase_4
```

Comment:
I kept them in the order they are taught. (As to why some are called unit_x and others phase_y is beyond me. Apparently they're on a long slow transition toward more consistent naming.) Also remember this is just a developer aid.